src/DataStructures/Tensor/EagerMath/DotProduct.hpp

```diff
@@ -75,6 +75,50 @@ Scalar<DataType> dot_product(
 }
 // @}
 
+// @{
+/*!
+ * \ingroup TensorGroup
+ * \brief Contract a vector or one-form with the first index of a tensor
+ *
+ * \details
+ * Returns \f$v_a T^a_{b\ldots}\f$, where \f$v_a\f$ is `vector_left`, and
+ * \f$T^a_{b\ldots}\f$ is `tensor_right`.
+ */
+template <typename DataType, typename Index, typename Symm,
+          typename... RemainingIndices,
+          typename ResultTensor = Tensor<DataType, tmpl::pop_front<Symm>,
+                                         index_list<RemainingIndices...>>>
+void dot_product(
```

Comment:
rename to `normal_dot_flux` (we are assuming this is what you want it for...) and move to an appropriate location,
and then it is always a one-form contracted into a tensor